### PR TITLE
test(regress): enable more bool tests except ERROR/IS UNKNOWN

### LINKED ISSUE #2535: Implement TEXT/STRING type as  alias of VARCHAR?
It's supported on legacy frontend.

Need some investigation.

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Compared `expected/boolean.out` between our version and PostgreSQL original one.
* Enable 2 more tests.
* Treat `text` and `varchar` as the same type. `text` is used frequently in regress tests.
  * Fixes #2535
* `IS UNKNOWN` is updated to `IS NULL` temporarily.
  * Tracked by #112
* All remaining mismatches (test cases commented out) are about testing error cases, where we are not able to match PostgreSQL's error message exactly.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

* SQL commands, functions, and operators

### Release note

`text` can now be specified as a type name. It is the same as `varchar`. (TBH it is more proper to say we implemented `text` and accept `varchar` as its alias, rather than to say we support `varchar`-without-length-limit and `text` is an alias.)

### CODE REVIEW

File: src/tests/regress/data/expected/boolean.out [hunk 1]
```diff
@@ -423,25 +423,25 @@ SELECT f1
...
+INSERT INTO BOOLTBL3 (d, b, o) VALUES ('false', false, 2);
+INSERT INTO BOOLTBL3 (d, b, o) VALUES ('null', null, 3);
+SELECT
+    d,
+    b IS TRUE AS istrue,
+    b IS NOT TRUE AS isnottrue,
+    b IS FALSE AS isfalse,
+    b IS NOT FALSE AS isnotfalse,
+    b IS NULL AS isunknown,
+    b IS NOT NULL AS isnotunknown
```
Comment (xiangjinwu) [inline src/tests/regress/data/expected/boolean.out:437]: > Notice that IS UNKNOWN and IS NOT UNKNOWN are effectively the same as IS NULL and IS NOT NULL, respectively, except that the input expression must be of Boolean type.

[ref](https://www.postgresql.org/docs/current/functions-comparison.html#:~:text=Notice%20that%20IS%20UNKNOWN%20and%20IS%20NOT%20UNKNOWN%20are%20effectively%20the%20same%20as%20IS%20NULL%20and%20IS%20NOT%20NULL%2C%20respectively%2C%20except%20that%20the%20input%20expression%20must%20be%20of%20Boolean%20type.)

File: src/tests/regress/src/schedule.rs [hunk 2]
```diff
@@ -478,7 +478,7 @@ where
             last_line_is_empty = true;
             last_is_select = false;
         } else if skip == 0 {
-            if line.to_lowercase().starts_with("select ") {
+            if line.to_lowercase().starts_with("select") {
```
Comment (xiangjinwu) [inline src/tests/regress/src/schedule.rs:481]: Some regress tests starts with `select\n`, `select(select` or `select;`.

**Review by fuyufjh — APPROVED** [review]

**Review by lmatz — APPROVED** [review]
LGTM, thanks!

**Comment by lmatz** [comment]
@mergifyio requeue

**Pull request merged**